client/src/components/RelatedProducts/Cards/AllCards.jsx

```diff
@@ -1,27 +1,21 @@
+/* eslint-disable max-len */
 import React from 'react';
 import styled from 'styled-components';
 import { Individualcard } from './Individualcard';
 import { useProd } from './ProdContext';
 import { useThumbnail } from './thumbnailContext';
 
 function AllCards() {
-  const relatedProds = useProd();
+  const relatedProds = useProd().relatedInformation;
   const thumbnail = useThumbnail();
+  const ratings = useProd().ratingsMeta;
+  // console.log(thumbnail, 'this is thumbnail');
 
   const zippedArray = [];
-  // if (!thumbnail.data) {
-  //   console.log('loading thumbnail');
-  // } if (!relatedProds.data) {
-  //   console.log('loading related prods');
-  // } else {
-  //   relatedProds.data.map((item, i) => {
-  //     zippedArray.push([item.data, thumbnail.data[i].data.results[0].photos[0].thumbnail_url]);
-  //   });
-  // }
-
-  if (thumbnail.data && relatedProds.data) {
+
+  if (thumbnail.data && relatedProds.data && ratings) {
     relatedProds.data.map((item, i) => {
-      zippedArray.push([item.data, thumbnail.data[i].data.results[0].photos[0].thumbnail_url]);
+      zippedArray.push([item.data, thumbnail.data[i].data.results[0].photos[0].thumbnail_url, ratings.data[i].data]);
```

Comment:
The second inout seems to be a long one, recommend break this down a bit (save part to an easy to understand const name) to make the code easier to understand. Also seems a bit strange to be pulling from first item for both result and photo in a map.

Reply:
The modal window is displayed when you click on the star, maybe I'll adjust the size of the star when I start the fine tuning of CSS